pkg/reconciler/taskrun/validate_taskrun.go

```diff
@@ -30,58 +30,41 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func validateParams(ctx context.Context, paramSpecs []v1beta1.ParamSpec, params []v1beta1.Param, matrix *v1beta1.Matrix) error {
+func validateParams(ctx context.Context, paramSpecs []v1beta1.ParamSpec, params v1beta1.Params, matrixAllParams v1beta1.Params) error {
 	neededParamsNames, neededParamsTypes := neededParamsNamesAndTypes(paramSpecs)
-	var matrixParams []v1beta1.Param
-	if matrix != nil {
-		matrixParams = matrix.Params
-	}
-	providedParamsNames := providedParamsNames(append(params, matrixParams...))
+	providedParams := params
+	providedParams = append(providedParams, matrixAllParams...)
```

Comment:
this looks like it modifies the input; can you double check that this function does not mutate its inputs?

Reply:
I don't believe it does

Reply:
my bad-- thinking of how things work in some other languages :(